### PR TITLE
fix: Fix L0_perf_nomodel shared memory

### PR DESCRIPTION
**Pull request by kthui**
#### What does the PR do?
Fix L0_perf_nomodel shared memory test - system and cuda.
* Update flag name `BENCHMARK_TEST_SHARED_MEMORY` to `TEST_SHARED_MEMORY`.
* Correctly set the `--output-shared-memory-size` flag for perf analyzer.

Root cause:
* The CI runners were re-configured to set `TEST_SHARED_MEMORY` instead of `BENCHMARK_TEST_SHARED_MEMORY`, causing the system/CUDA shared memory tests to run as non-shared memory test, resulting in no data reported for both shared memory tests since the re-configuration.

#### Checklist
- [x] PR title reflects the change and is of format `<commit_type>: <Title>`
- [x] Changes are described in the pull request.
- [x] Related issues are referenced.
- [x] Populated [github labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels) field
- [x] Added [test plan](#test-plan) and verified test passes.
- [x] Verified that the PR passes existing CI.
- [x] Verified copyright is correct on all changed files.
- [ ] Added _succinct_ git squash message before merging [ref](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
- [x] All template sections are filled out.
- [ ] Optional: Additional screenshots for behavior/output changes with before/after.

#### Commit Type:
Check the [conventional commit type](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type)
box here and add the label to the github PR.
- [ ] build
- [ ] ci
- [ ] docs
- [ ] feat
- [x] fix
- [ ] perf
- [ ] refactor
- [ ] revert
- [ ] style
- [ ] test

#### Related PRs:
N/A

#### Where should the reviewer start?
N/A

#### Test plan:
N/A

- CI Pipeline ID: 19429907

#### Caveats:
N/A

#### Background
N/A

#### Related Issues: (use one of the action keywords Closes / Fixes / Resolves / Relates to)
N/A

### CODE REVIEW

**Review by indrajit96 — COMMENTED** [review]
Can you add the RCA of tests not completing either on the bug or PR description?

File: qa/L0_perf_nomodel/test.sh [hunk 1]
```diff
@@ -72,14 +72,14 @@ RUNTEST=./run_test.sh
 # by 4.
 TENSOR_SIZE_16MB=$((4*1024*1024))
 
-if [ "$BENCHMARK_TEST_SHARED_MEMORY" == "system" ]; then
+if [ "$TEST_SHARED_MEMORY" == "system" ]; then
```
Comment (indrajit96) [inline qa/L0_perf_nomodel/test.sh:75]: Why did we need to rename?

Comment (indrajit96) [reply qa/L0_perf_nomodel/test.sh:75]: NVM, Description explains it

**Comment by indrajit96** [comment]
Just a FYI, we might need to edit these test soon one the PA fixes for C API and PA Hang are done.
LGTM for now. Approving,

**Review by indrajit96 — APPROVED** [review]

**Comment by kthui** [comment]
I will merge to main first. We can file additional tickets if there are breakages with the PA fixes.

**Pull request merged**